internal/services/recommender.go

```diff
@@ -75,6 +75,19 @@ func ProcessEvent(msg *kafka.Message) {
 				} else {
 					log.Infof("Recommendation saved for experiment - %s and end_interval - %s", kafkaMsg.Experiment_name, recommendationSet.MonitoringEndTime)
 				}
+
+				// Create entry into HistoricalRecommendationSet table.
+				historicalRecommendationSet := model.HistoricalRecommendationSet{
```

Comment:
In my opinion, no one really needs to care about this internal implementation as it is just creating copy of record into another table.
So, Will it make sense to extract out creation of recommendation set and historical set into a method so that it can be reuse to other places?

Reply:
To extract this logic to some other function we will need to export few internal type which we have declared under `/internal/type` this will create bit much refactoring of code. Can we do this later in some other PR?